core/src/main/java/com/sequenceiq/cloudbreak/fluent/FluentConfigService.java

```diff
@@ -26,78 +34,127 @@
 
     private static final String S3_PROVIDER_PREFIX = "s3";
 
+    private static final String WASB_PROVIDER_PREFIX = "wasb";
+
     private static final String DEFAULT_PROVIDER_PREFIX = "stdout";
 
     public FluentConfigView createFluentConfigs(Stack stack, Telemetry telemetry) {
+        final FluentConfigView.Builder builder = new FluentConfigView.Builder();
+        boolean fluentEnabled = false;
         if (telemetry != null && telemetry.getLogging() != null) {
             Logging logging = telemetry.getLogging();
             String clusterName = stack.getCluster().getName();
             String platform = stack.getCloudPlatform();
-            StackType stackType = stack.getType();
+            String storageLocation = logging.getStorageLocation();
 
-            String clusterType = CLUSTER_TYPE_DISTROX;
-            if (StackType.DATALAKE.equals(stackType)) {
-                clusterType = CLUSTER_TYPE_SDX;
-            }
-            String s3bucket = null;
-            String s3LogFolderName = null;
-            boolean fluentEnabled = false;
-            String providerPrefix = DEFAULT_PROVIDER_PREFIX;
+            String clusterType = StackType.DATALAKE.equals(stack.getType()) ? CLUSTER_TYPE_SDX : CLUSTER_TYPE_DISTROX;
+            String logFolderName = Paths.get(CLUSTER_LOG_PREFIX, clusterType, clusterName).toString();
+
+            builder.withPlatform(platform)
+                    .withOverrideAttributes(
+                            logging.getAttributes() != null ? new HashMap<>(logging.getAttributes()) : new HashMap<>()
```

Comment:
Alternatively: `java.util.Objects.requireNonNullElse​(new HashMap<>(logging.getAttributes()), new HashMap<>())`.

Reply:
i keep it that way for now, as im copying the map, i'm creating a new one and passing the attributes as a map, so before the null check in Object.xxx function, it will produce an NPE already.

Reply:
Oops, you are right. Was indeed a silly suggestion as one should really check for `logging.getAttributes()` explicitly. 😊